tests/geometry/test-aabb.cpp

```diff
@@ -0,0 +1,312 @@
+    // create a grid of separate boxes
+    std::vector< geode::BoundingBox< dimension > > box_vector =
+        create_box_vector< dimension >( nb_boxes, box_size );
+    AABBTree< dimension > aabb( box_vector );
+
+    // check number of box
+    OPENGEODE_EXCEPTION( aabb.nb_bboxes() == box_vector.size(),
+        "[Test] Error ... wrong number of box in the aabb tree" );
+    return;
+}
+
+template < index_t dimension >
+class BoxAABBEvalDistance
+{
+public:
+    BoxAABBEvalDistance(
+        const std::vector< geode::BoundingBox< dimension > >& bounding_boxes )
+        : bounding_boxes_( bounding_boxes )
+    {
+    }
+    ~BoxAABBEvalDistance() = default;
+
+    // EvalDistance
+    std::tuple< double, Point< dimension > > operator()(
+        const Point< dimension >& query, index_t curent_element_box ) const
+    {
+        Point< dimension > box_center =
+            ( bounding_boxes_[curent_element_box].min()
+                + bounding_boxes_[curent_element_box].max() )
+            / 2.;
+        Vector< dimension > vec( box_center, query );
+        return std::tuple< double, Point< dimension > >{ vec.length(),
+            box_center };
+    }
+
+private:
+    const std::vector< geode::BoundingBox< dimension > >& bounding_boxes_;
+};
+
+template < index_t dimension >
+void test_nearest_neighbor_search()
+{
+    index_t nb_boxes = 10;
+    double box_size = 0.75;
+    // create a grid of boxes that intersect
+    std::vector< geode::BoundingBox< dimension > > box_vector =
+        create_box_vector< dimension >( nb_boxes, box_size );
+    AABBTree< dimension > aabb( box_vector );
+
+    BoxAABBEvalDistance< dimension > disteval =
+        BoxAABBEvalDistance< dimension >( box_vector );
+
+    for( geode::index_t i : Range( nb_boxes ) )
+    {
+        for( index_t j : Range( nb_boxes ) )
+        {
+            Point< dimension > box_center;
+            box_center.set_value( 0, (double) i );
+            box_center.set_value( 1, (double) j );
+
+            Point< dimension > query;
+            // query point is in the up right corner of each box.
+            query.set_value( 0, (double) i + box_size / 2. );
+            query.set_value( 1, (double) j + box_size / 2. );
+            index_t box_id;
+            Point< dimension > neatest_point;
+            double distance;
+            std::tie( box_id, neatest_point, distance ) =
+                aabb.closest_element_box( query, disteval );
+
+            OPENGEODE_EXCEPTION( box_id == global_box_index( i, j, nb_boxes ),
+                "[Test] Error ... wrong nearest box index" );
+            OPENGEODE_EXCEPTION( neatest_point == box_center,
+                "[Test] Error ... wrong nearest point " );
+            Vector< dimension > vec( box_center, query );
+            OPENGEODE_EXCEPTION(
+                distance == vec.length(), "[Test] Error ... wrong distance " );
+        }
+    }
+}
+
+template < index_t dimension >
+class BoxAABBEvalIntersection
```

Comment:
Is `BoxAABBEvalIntersection` the right name since it checks box inclusions: `EvalBoxInclusion` ?

Reply:
the operator check inclusion after box intersections...

Reply:
What about BoxAABBIntersection

Reply:
it also evaluate the intersection between a box and the tree. Would you like two classes?